host/aes_crypto.c

```
@@ -134,106 +108,112 @@ static void ctr128_inc(uint8_t *counter, uint32_t increment)
 
 }
 
-static uint32_t commit_buffer_tee_aes_ctr128_decrypt(uint32_t sz,  const void* iv,
-    uint32_t iv_size,
-    const char* key, uint32_t key_size, int flags)
+static void free_mem(void)
 {
+  PR("Release IV shared memory...\n");
+  TEEC_ReleaseSharedMemory(&g_iv);
+  PR("Release key shared memory...\n");
+  TEEC_ReleaseSharedMemory(&g_key);
+}
+
+/* Decrypt buffer */
+
+int
+TEE_AES_ctr128_encrypt(const unsigned char* in_data,
+    unsigned char* out_data,
+    uint32_t length, const char* key,
+    unsigned char iv[CTR_AES_BLOCK_SIZE],
+    unsigned char ecount_buf[CTR_AES_BLOCK_SIZE],
+    unsigned int *num,
+    uint32_t offset,
+    bool secure) {
   TEEC_Result res;
+  int secure_fd = -1;
   TEEC_Operation op;
   uint32_t err_origin;
+  uint32_t n = 0;
+  uint32_t blockOffset = *num;
+  uint32_t len = length;
 
-  assert(key_size == g_key.size);
-  assert(iv_size == CTR_AES_IV_SIZE);
+  if (!key || !out_data || !num || !iv)
+    return 0;
```

Do you want to return an error?

---

will fix